### ISSUE TITLE
Revival

### ISSUE THREAD

**Issue by gaborcsardi** [richfitz/pathr#4]
I need some of these over and over again, so let's finish this package. Or least I'll add the things I need. FYI.

**Comment by richfitz** [comment]
OK, cool.  Want me to add you as a collaborator and/or move somewhere author neutral such as ropenscilabs?

I also have periodic need for this so am open to spending a bit of time on it (but possibly not much spare bandwidth over the next 2-3 weeks).

**Comment by gaborcsardi** [comment]
I don't mind, I am fine wil PRs as well for now. :)

**Comment by gaborcsardi** [comment]
But then you need to review them. :)

**Comment by richfitz** [comment]
I'll review them quickly, don't worry.

**Comment by gaborcsardi** [comment]
Quick q. I need some regular expression matching for `path_isabs` on windows. Do you mind if I depend on this: https://github.com/MangoTheCat/rematch ? It is small, R-only and has no dependencies.

EDIT: and I just submitted it to CRAN....

**Comment by richfitz** [comment]
Go for it!  Looks like it might be useful in general though the package.

**Comment by jennybc** [comment]
I didn't even know I was watching this repo until the sudden re-awakening, but I am so glad to learn of rematch! I [just wrote this](https://github.com/jennybc/cellranger/blob/582c4d2511802964ac0bca39875f9e8f8f87bbbb/R/utils.R#L52-L61) and will gladly rip it back out again :).

**Comment by gaborcsardi** [comment]
@jennybc Yeah, right? :) I wrote this about ~5 times at various places, and today I decided that it was enough. It is hard to get all the corner cases right, so it will be just perfect in its own package.

Btw. `stringr::str_match` is also pretty good, if you need sg comprehensive.

**Comment by jennybc** [comment]
Can you name a non-capturing group? _Yes I googled it and ... answer not clear, so am lazy-githubbing with you._

**Comment by gaborcsardi** [comment]
@jennybc I don't think you can in PCRE. It is not in the docs, and the obvious `(?:<name>...)` syntax does not work.

**Comment by jennybc** [comment]
@gaborcsardi That was my conclusion. It seems unfortunate. I don't see any reason these two things have to be incompatible. Yet I believe they are.

**Comment by jennybc** [comment]
I see I have high-jacked this issue before, but now I'm here to make an on-topic comment.

pathr would be really useful, so count me as another 👍!

I'm particularly interested in `path_rel()`. If user gives me "foo/bar/blah.R", I might have to normalize it into an absolute path and then form, say, a related path = a normalized version of "foo/bar/blah.md". But I'd like to message that in a way that matches the path style the user used, not the awful long normalized version. It feels like pathr would do the necessary file path arithmetic.

**Comment by gaborcsardi** [comment]
@jennybc I might misunderstand sg, but `path_rel()` already does what you want, no?

**Comment by jennybc** [comment]
Yes! I just meant that `path_rel()` would be super useful and therefore nice to have on CRAN.

**Comment by richfitz** [comment]
update: this is my current train project (as I need something small and self contained to work on at the moment) and so will see a series of updates over the next few days/weeks

**Comment by gaborcsardi** [comment]
\o/

**Comment by jennybc** [comment]
🚆🎉
